infracheck/checks/domain-expiration

```diff
@@ -47,43 +56,78 @@ class DomainCheck:
         if not self._domain or self._domain == '':
             raise Exception('Domain must be specified')
 
-    def _check_with_wait(self):
+    def _check_with_wait(self) -> WhoisResult:
         time.sleep(1)
         max_retries = 10
         retries = 0
 
-        while True:
+        while retries <= max_retries:
             try:
-                query = self.whois.query(self._domain)
+                query = self._ask_whois_library(self._domain)
 
                 if not query or not query.expiration_date:
                     return self._parse_shell_whois(self._domain)
 
                 return query
+
             except Exception as e:
                 retries += 1
 
                 if retries > max_retries:
                     raise e
 
+                if isinstance(e, whois.exceptions.UnknownTld):
+                    shell_check_result = self._parse_shell_whois(self._domain)
+
+                    if not shell_check_result or not shell_check_result.expiration_date:
+                        continue
+
+                    return shell_check_result
+
                 if "request limit exceeded" in str(e):
                     time.sleep(5)
 
-    def _parse_shell_whois(self, domain: str):
+    def _ask_whois_library(self, domain: str) -> WhoisResult:
+        _stdout_copy = sys.stdout
+
+        # the "whois" library is using print(), sys.stdout needs to be mocked for a moment to not print those
+        # messages
+        with open('/dev/null', 'w') as null_out:
+            try:
+                sys.stdout = null_out
+
+                return self.whois.query(domain)
+            finally:
+                sys.stdout = _stdout_copy
+
+    @staticmethod
+    def _ask_whois_in_shell(domain: str) -> str:
```

Comment:
What's happening here? How does output make it to the shell? Is it printed out by this library/class?

Reply:
Or is it a call to `whois` in terminal?

Reply:
It is a spawned shell call, the output is captured from shell.